core/cloudflow-operator/src/test/scala/cloudflow/operator/runner/SparkRunnerSpec.scala

```diff
@@ -115,6 +115,169 @@ class SparkRunnerSpec extends WordSpecLike with OptionValues with MustMatchers w
+      crd.spec.executor.labels.get("key2") mustBe None
+    }
+
+    "read from config custom labels and add them to the executor pod's spec" in {
+
+      val crd = SparkRunner.resource(
+        deployment = deployment,
+        app = app,
+        configSecret = Secret(
+          metadata = ObjectMeta(),
+          data = Map(
+            cloudflow.operator.event.ConfigInputChangeEvent.PodsConfigDataKey ->
+                """
+                |kubernetes.pods.executor {
+                | labels: {
+                |    "key1" : "value1",
+                |    "key2" : "value2"
+                |          }
+                | containers.container {
+                |  env = [
+                |    {
+                |      name = "FOO"
+                |      value = "BAR"
+                |    }
+                |   ]
+                |}
+                |}
+                """.stripMargin.getBytes()
+          )
+        ),
+        namespace = namespace
+      )
+
+      crd.spec.executor.labels.get("key1") mustBe Some("value1")
+      crd.spec.driver.labels.get("key1") mustBe None
+      crd.spec.executor.labels.get("key2") mustBe Some("value2")
+      crd.spec.driver.labels.get("key2") mustBe None
+    }
+
+    "read from config custom labels and add them to the driver and executor pods specs" in {
+
+      val crd = SparkRunner.resource(
+        deployment = deployment,
+        app = app,
+        configSecret = Secret(
+          metadata = ObjectMeta(),
+          data = Map(
+            cloudflow.operator.event.ConfigInputChangeEvent.PodsConfigDataKey ->
+                """
+                |kubernetes.pods.pod {
+                | labels: {
+                |    "key1" : "value1",
+                |    "key2" : "value2"
+                |          }
+                | containers.container {
+                |  env = [
+                |    {
+                |      name = "FOO"
+                |      value = "BAR"
+                |    }
+                |   ]
+                |}
+                |}
+                """.stripMargin.getBytes()
+          )
+        ),
+        namespace = namespace
+      )
+
+      crd.spec.driver.labels.get("key1") mustBe Some("value1")
+      crd.spec.executor.labels.get("key1") mustBe Some("value1")
+      crd.spec.driver.labels.get("key2") mustBe Some("value2")
+      crd.spec.executor.labels.get("key2") mustBe Some("value2")
+    }
+
+    "read from config DIFFERENT custom labels and add them to the driver and executor pods specs" in {
+
+      val crd = SparkRunner.resource(
+        deployment = deployment,
+        app = app,
+        configSecret = Secret(
+          metadata = ObjectMeta(),
+          data = Map(
+            cloudflow.operator.event.ConfigInputChangeEvent.PodsConfigDataKey ->
+                """
+                |kubernetes.pods {
+                |  driver {
+                |    labels: {
+                |       "key1" : "value1",
```

Comment:
do `key = value` instead in hocon style?

Reply:
done